### PR TITLE
Add feature to exclude matches based on configurable regex

### PR DESCRIPTION
**Pull request by JasperCraeghs**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -293,20 +293,42 @@ path to configuration file
...
+
+.. code-block:: json
+
+    {
+        "sphinx":{
+    	    "enabled": true,
+    	    "min": 0,
+    	    "max": 0,
+            "exclude": "RemovedInSphinx\\d+Warning"
+        }
```
Comment (Letme) [inline README.rst]: Alignment here, but because of that can we make it array of strings for exclude regexes?

File: src/mlx/warnings.py [hunk 2]
```diff
@@ -173,8 +173,13 @@ def config_parser_json(self, config):
             try:
                 if bool(config[checker.name]['enabled']):
                     self.activate_checker(checker)
-                    self.get_checker(checker.name).set_maximum(int(config[checker.name]['max']))
-                    self.get_checker(checker.name).set_minimum(int(config[checker.name]['min']))
+                    checker.set_maximum(int(config[checker.name]['max']))
+                    checker.set_minimum(int(config[checker.name]['min']))
```
Comment (Letme) [inline src/mlx/warnings.py]: I would say self.get_checker functtion is there to protect agains i think unexsmisting checkers? I would keep it that way...

File: README.rst (hunk 1 above)
Comment (JasperCraeghs) [reply README.rst]: It is possible, but a user can use `|` OR-logic inside this single regex to achieve the same result, right?

Comment (Letme) [reply README.rst]: True, but it is less readable that way since if you want to exclude 5 different error messages you will have a lot of brackets (especially with more complex error parsing).

File: src/mlx/warnings.py (hunk 2 above)
Comment (JasperCraeghs) [reply src/mlx/warnings.py]: The `checker` variable is an element from the `WarningsPlugin.public_checkers` list. It has been "activated" in a line above like so: `self.activate_checker(checker)`. Using the `self.get_checker` function merely tests the functionality of the `self.activate_checker` function in this use case.

File: README.rst (hunk 1 above)
Comment (JasperCraeghs) [reply README.rst]: It must now be a list: df6e6440110e075724cf6a44d9b7781b62d27f9b

File: src/mlx/warnings.py (hunk 2 above)
Comment (Letme) [reply src/mlx/warnings.py]: yes, but that is the case. Because at this moment in time we use activated checker and with current change we blindly just use any checker as input parameter. Strangely the unit tests did not fail...

File: src/mlx/warnings_checker.py [hunk 3]
```diff
@@ -121,32 +134,63 @@ def return_check_limits(self):
...
+        '''
+        if self.verbose:
+            print(message)
+
 
 class RegexChecker(WarningsChecker):
     name = 'regex'
     pattern = None
 
-    def __init__(self, verbose=False):
```
Comment (Letme) [inline src/mlx/warnings_checker.py]: Why was this removed? I think this enabled us to be verbose inside the checkers...

File: tests/test_config.py [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
-        self.assertEqual(warnings.return_count(), 1)
+        warnings = WarningsPlugin(verbose=True, config_file="tests/config_example.json")
+        with patch('sys.stdout', new=StringIO()) as verbose_output:
+            warnings.check('testfile.c:6: warning: group test: ignoring title "Some test functions" that does not match old title "Some freaky test functions"')
+            self.assertEqual(warnings.return_count(), 0)
+            warnings.check('<testcase classname="dummy_class" name="dummy_name"><failure message="some random message from test case" /></testcase>')
+            self.assertEqual(warnings.return_count(), 0)
+            deprecation_warning = 'sphinx/application.py:402: RemovedInSphinx20Warning: app.info() is now deprecated. Use sphinx.util.logging instead.'
+            warnings.check(deprecation_warning)
+            self.assertEqual(warnings.return_count(), 0)  # ignored because of configured "exclude" regex
```
Comment (Letme) [inline tests/test_config.py]: Can we duplicate this test where we add additional `config_example_exclude.json`, just so that we can confirm functionality without exclude regex for all these checkers.

File: src/mlx/warnings.py (hunk 2 above)
Comment (JasperCraeghs) [reply src/mlx/warnings.py]: This code is only reached when the value for key `"enabled"` is `True`. The elements in `public_checkers` list are initialized class objects that get added to `activated_checkers` dict by `self.activate_checker(checker)`.  No new WarningsChecker objects get instantiated here.

Comment (Letme) [reply src/mlx/warnings.py]: True. With current flow the checker needs to be enabled before this line is executed anyway.

File: tests/test_config.py (hunk 4 above)
Comment (JasperCraeghs) [reply tests/test_config.py]: Okay, here I added back the original unit test: cf767ff94eda158eb512d33d33069c5dc104acce. The exclude key is completely optional. It can be left out or have an empty list `[]` as value.

**Comment by Letme** [comment]
Maybe as a side note. Can we add something like `--exclude-sphinx-deprecation` flag to the warnings-plugin executeable which adds this fixed excludes to the checker configuration when used in command line? I would still rather change a default and have `--include-sphinx-deprecation` as we could see this is now hurting us a bit.

**Comment by JasperCraeghs** [comment]
Are you not completely happy with the configuration-file-only approach of the feature? Do you want both `--exclude-sphinx-deprecation` and `--include-sphinx-deprecation` flags? What do you mean by _change a default_ and _this is now hurting us a bit_? In any case adding an extra flag will happen in a separate MR so this can be released if you want.

**Comment by Letme** [comment]
This deprecstion warnings that now count is what basically triggers a warning when something out of our project scope is having a warning. It means we coubt warnings which are not directly related to project warnings but to infrastructure.  So by default these should not count, but should be enabled by flag...
